aptos-move/block-executor/src/liftings.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Implementation of sync and unsync versions of data structures which are used
+//! to process aggregator liftings.
+//!
+//! Both have:
+//!   1. A counter used to generate a new unique (per-block) identifier.
+//!   2. A map from identifiers to lifted Move values.
+
+use dashmap::{mapref::one::Ref, DashMap};
+use move_vm_types::values::Value;
+use std::{
+    collections::HashMap,
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+pub(crate) struct UnsyncLiftings {
+    counter: u64,
```

Comment:
does this counter have a separate namespace from aggregator context generating ids? or do we need an Arc and a shared counter?

it's not only sync part that generates ids during native calls

Reply:
I believe the aggregator context should also be using this counter for generating ids. We need a way of sharing the counter via traits.

Reply:
Yep, there is `increment` method which should be used by AggregatorResolver trait